subcommands/targets/offline-update.go

```diff
@@ -0,0 +1,264 @@
+package targets
+
+import (
+	"archive/tar"
+	"compress/bzip2"
+	"fmt"
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"io"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	factory string
+	prod    bool
+	tag     string
+)
+
+func init() {
+	offlineUpdateCmd := &cobra.Command{
+		Use:   "offline-update <target-name> <dst>",
+		Short: "Download Target content for an offline update",
+		Run:   doOfflineUpdate,
+		Args:  cobra.ExactArgs(2),
+	}
+	cmd.AddCommand(offlineUpdateCmd)
+	offlineUpdateCmd.Flags().BoolVarP(&prod, "prod", "", false,
+		"Instruct to fetch content of production Target")
+	offlineUpdateCmd.Flags().StringVarP(&tag, "tag", "", "master",
+		"Target tag")
```

Comment:
The tag parameter is the key here so I'd recommend to make it mandatory. The doesn't just fetch "any" targets, it fetches targets for the specific tag. In particular, for prod targets different tags may contain slightly different target items of the same name; and they are almost 100% of the time different from the CI targets of the same name.

Reply:
Yep, it makes sense to make the `tag` param mandatory.